### ISSUE TITLE
[iOS, Xcode] Upload Debug Files fails on HTTP error 4xx

### ISSUE THREAD

**Issue by thefranfran** [getsentry/sentry-react-native#3551]
**OS:**
- []  Windows
- [X]  MacOS
- []  Linux

**Platform:**
- [X]  iOS
- []  Android

**SDK:**
- [X]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:** 5.17.0

**`react-native` version:** 0.73.2

**Are you using Expo?**
- []  Yes
- [X]  No

**Are you using sentry.io or on-premise?**
- [X]  sentry.io (SaaS)
- []  on-premise

**Configuration:**

(`@sentry/react-native`)
```
Sentry.init({
  dsn: 'https://...@sentry.io/...'
  // other options
});
```
---
**I have the following issue:**

I installed the package with command `npx @sentry/wizard@latest -s -i reactNative`
and the metro config is setup aswell
It's look like it's an error during the execution of the Run Script Phase which is `Upload Debug Symbols to Sentry`
And the script is `/bin/sh ../node_modules/@sentry/react-native/scripts/sentry-xcode-debug-files.sh`

**Steps to reproduce:**
- `npx @sentry/wizard@latest -s -i reactNative`
- Build project with XCode (15.0.1)

**Actual result:**
Command PhaseScriptExecution failed with a nonzero exit code
PhaseScriptExecution Upload\ Debug\ Symbols\ to\ Sentry 

[BuildFail.txt](https://github.com/getsentry/sentry-react-native/files/14050771/BuildFail.txt)

**Comment by andrewdavidcostello** [comment]
I think we are also encountering this issue, the problem seems to be a difference in paths.

`Error: ENOENT: no such file or directory, open`

For example:

`HBC_SOURCEMAP_FILE=/Users/expo/Library/Developer/Xcode/DerivedData/[redacted]/Build/Intermediates.noindex/ArchiveIntermediates/[redacted]/BuildProductsPath/Release-iphoneos/[redacted].app/main.jsbundle.map`

but then it tries to load from

`/Users/expo/Library/Developer/Xcode/DerivedData/[redacted]/Build/Intermediates.noindex/ArchiveIntermediates/[redacted]/BuildProductsPath/Release-iphoneos/main.jsbundle.map`

**Comment by krystofwoldrich** [comment]
Hi @thefranfran,
thank you for the message and the logs, sadly we don't se the cause of the crash in the logs.

Could you run the build again with this environmental variable and share the new logs?

```bash
export SENTRY_CLI_EXTRA_ARGS="--force-foreground --log-level=debug"
```

**Comment by thefranfran** [comment]
Hi @krystofwoldrich, 

Thank you for your feedback
I'd put the extra args as you mentioned in the Phase Script **Upload Debug Symbols to Sentry**

`#!/bin/bash
# Upload Debug Symbols to Sentry Xcode Build Phase
# PWD=ios

# print commands before executing them and stop on first error
set -x -e

[ -z "$WITH_ENVIRONMENT" ] && WITH_ENVIRONMENT="../node_modules/react-native/scripts/xcode/with-environment.sh"

if [ -f "$WITH_ENVIRONMENT" ]; then
  # load envs if loader file exists (since rn 0.68)
  . "$WITH_ENVIRONMENT"
fi

LOCAL_NODE_BINARY=${NODE_BINARY:-node}

[ -z "$SENTRY_PROPERTIES" ] && export SENTRY_PROPERTIES=sentry.properties

[ -z "$SENTRY_CLI_EXECUTABLE" ] && SENTRY_CLI_PACKAGE_PATH=$("$LOCAL_NODE_BINARY" --print "require('path').dirname(require.resolve('@sentry/cli/package.json'))")
[ -z "$SENTRY_CLI_EXECUTABLE" ] && SENTRY_CLI_EXECUTABLE="${SENTRY_CLI_PACKAGE_PATH}/bin/sentry-cli"

[[ $SENTRY_INCLUDE_NATIVE_SOURCES == "true" ]] && INCLUDE_SOURCES_FLAG="--include-sources" || INCLUDE_SOURCES_FLAG=""

EXTRA_ARGS="--force-foreground --log-level=debug $SENTRY_CLI_DEBUG_FILES_UPLOAD_EXTRA_ARGS $INCLUDE_SOURCES_FLAG"

UPLOAD_DEBUG_FILES="\"$SENTRY_CLI_EXECUTABLE\" debug-files upload $EXTRA_ARGS \"$DWARF_DSYM_FOLDER_PATH\""

if [ "$SENTRY_DISABLE_AUTO_UPLOAD" != true ]; then
  /bin/sh -c "\"$LOCAL_NODE_BINARY\" $UPLOAD_DEBUG_FILES"
else
  echo "SENTRY_DISABLE_AUTO_UPLOAD=true, skipping debug files upload"
fi`

I do not know if it will help you so tell me if it doesn't 
[Build 2024-01-29T16-41-19.txt](https://github.com/getsentry/sentry-react-native/files/14086478/Build.2024-01-29T16-41-19.txt)

**Comment by krystofwoldrich** [comment]
@thefranfran It seems like the output didn't change, how does your `.xcode.env` and `.xcode.env.local` looks like, it seem like the output log ends on the line `NODE_BINARY=`.

**Comment by thefranfran** [comment]
@krystofwoldrich 

my `.xcode.env` is

export NODE_BINARY=$(command -v node)
export SENTRY_CLI_EXTRA_ARGS="--force-foreground --log-level=debug"

and my `.xcode.env.local` is

export NODE_BINARY=/opt/homebrew/bin/node
export SENTRY_CLI_EXTRA_ARGS="--force-foreground --log-level=debug"

**Comment by krystofwoldrich** [comment]
@thefranfran Could you try to run the build without `export NODE_BINARY=$(command -v node)`?

**Comment by thefranfran** [comment]
@krystofwoldrich  It looks like the same output

+ '[' -z '' ']'
+ WITH_ENVIRONMENT=../node_modules/react-native/scripts/xcode/with-environment.sh
+ '[' -f ../node_modules/react-native/scripts/xcode/with-environment.sh ']'
+ . ../node_modules/react-native/scripts/xcode/with-environment.sh
+++ command -v node
++ NODE_BINARY=
Command PhaseScriptExecution failed with a nonzero exit code

**Comment by krystofwoldrich** [comment]
Are you able to reproduce this is a new app? Could you create a minimal reproducible example?

**Comment by thefranfran** [comment]
Yes so I did the steps from the react native documentation (0.73.3)

`npx react-native@latest init AwesomeProject`
`cd ios to navigate to the ios folder.`
`bundle install to install [Bundler](https://bundler.io/)`
`bundle exec pod install to install the iOS dependencies managed by CocoaPods.`

I am able to build it and run it on Iphone 15 (17.2)
And my `.xcode.env` have the `export NODE_BINARY=$(command -v node)`

So now I run the command `npx @sentry/wizard@latest -s -i reactNative`
I installed the pods, clean project and I have the same output 

+ '[' -z '' ']'
+ WITH_ENVIRONMENT=../node_modules/react-native/scripts/xcode/with-environment.sh
+ '[' -f ../node_modules/react-native/scripts/xcode/with-environment.sh ']'
+ . ../node_modules/react-native/scripts/xcode/with-environment.sh
+++ command -v node
++ NODE_BINARY=
Command PhaseScriptExecution failed with a nonzero exit code

The error is located in the script `/bin/sh ../node_modules/@sentry/react-native/scripts/sentry-xcode-debug-files.sh` exec in the Build Phases `Upload Debug Symbols to Sentry` which execute the react-native script and check the `WITH_ENVIRONMENT` 

[ -z "$WITH_ENVIRONMENT" ] && WITH_ENVIRONMENT="../node_modules/react-native/scripts/xcode/with-environment.sh"

if [ -f "$WITH_ENVIRONMENT" ]; then
  # load envs if loader file exists (since rn 0.68)
  . "$WITH_ENVIRONMENT"
fi

LOCAL_NODE_BINARY=${NODE_BINARY:-node}

may be I can force the node path here ? 
or can we skip the WITH_ENVIRONMENT assignment ?

**Comment by soadtechdev** [comment]
Any solution?
The first run works and registers the crash in sentry
But when I try to run the app again it generates the error and does not compile any more

```
+ '[' -z '' ']'
+ WITH_ENVIRONMENT=../node_modules/react-native/scripts/xcode/with-environment.sh
+ '[' -f ../node_modules/react-native/scripts/xcode/with-environment.sh ']''
+ ../node_modules/react-native/scripts/xcode/with-environment.sh
+++ command -v node
++ NODE_BINARY=
PhaseScriptExecution command failed with non-zero exit code.
```

**Comment by fredriksormo** [comment]
Any updates?

I have the exact same issue. If you need a temp solution to be able to run your build on simulator you can just enable the script for install builds only.

![image](https://github.com/getsentry/sentry-react-native/assets/21368571/67fdca48-673c-4097-8e73-558423bc0ba2)


But if you are trying to build for release I don't have any solution yet.

**Comment by krystofwoldrich** [comment]
Hello everyone, 
here is a fix:
- https://github.com/getsentry/sentry-react-native/pull/3573

Until the hotfix is out, you can use `patch-package` or manually edit the `sentry-xcode-debug-files.sh` script.

**Comment by arapocket** [comment]
I have the same issue, used the steps here
https://docs.expo.dev/guides/using-sentry/#example-configuration-that-instruments-sentry-for-expo

**Comment by aminos02** [comment]
i faced the same issue , and i fixed that by modifying the project.pbxproj file 
16A9994D004C413890C6E178 /* Upload Debug Symbols to Sentry */ = {
			....
			runOnlyForDeploymentPostprocessing = 1;   // change 0 to 1
			....
		};

**Comment by alterx** [comment]
I'm having the same issue as @arapocket 
I'm moving away from the old `sentry-expo` package. I have followed these [instructions](https://docs.expo.dev/guides/using-sentry/) but now my iOS build (local and in EAS) is failing 

```
❌  error: API request failed


[RUN_FASTLANE] ⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » mUnityStaging/mUnityStaging » Build Phases » 'Upload Debug Symbols to Sentry'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script

[RUN_FASTLANE] ⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » mUnityStaging/mUnityStaging » Build Phases » '[CP-User] [RNFB] Core Configuration'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script

[RUN_FASTLANE] ⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » mUnityStaging/mUnityStaging » Build Phases » '[CP-User] [RNFB] Crashlytics Configuration'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script

[RUN_FASTLANE]     Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run
[RUN_FASTLANE] ▸ ** ARCHIVE FAILED **
[RUN_FASTLANE] ▸ The following build commands failed:
[RUN_FASTLANE] ▸        PhaseScriptExecution Bundle\ React\ Native\ code\ and\ images /Users/carlosvega/Library/Developer/Xcode/DerivedData/mUnityStaging-ddududtjqwkahchdtkusacvqoxmy/Build/Intermediates.noindex/ArchiveIntermediates/mUnityStaging/IntermediateBuildFilesPath/mUnityStaging.build/Release-iphoneos/mUnityStaging.build/Script-00DD1BFF1BD5951E006B06BC.sh (in target 'mUnityStaging' from project 'mUnityStaging')
[RUN_FASTLANE] ▸ (1 failure)
[RUN_FASTLANE] ** ARCHIVE FAILED **
[RUN_FASTLANE] 
[RUN_FASTLANE] 
[RUN_FASTLANE] The following build commands failed:
[RUN_FASTLANE]  PhaseScriptExecution Bundle\ React\ Native\ code\ and\ images /Users/carlosvega/Library/Developer/Xcode/DerivedData/mUnityStaging-ddududtjqwkahchdtkusacvqoxmy/Build/Intermediates.noindex/ArchiveIntermediates/mUnityStaging/IntermediateBuildFilesPath/mUnityStaging.build/Release-iphoneos/mUnityStaging.build/Script-00DD1BFF1BD5951E006B06BC.sh (in target 'mUnityStaging' from project 'mUnityStaging')
[RUN_FASTLANE] (1 failure)
[RUN_FASTLANE] Exit status: 65
[RUN_FASTLANE] 
[RUN_FASTLANE] +-------------+-------------------------+
[RUN_FASTLANE] |           Build environment           |
[RUN_FASTLANE] +-------------+-------------------------+
[RUN_FASTLANE] | xcode_path  | /Applications/Xcode.app |
[RUN_FASTLANE] | gym_version | 2.212.1                 |
[RUN_FASTLANE] | sdk         | iPhoneOS17.4.sdk        |
[RUN_FASTLANE] +-------------+-------------------------+
[RUN_FASTLANE] Looks like fastlane ran into a build/archive error with your project
[RUN_FASTLANE] It's hard to tell what's causing the error, so we wrote some guides on how
[RUN_FASTLANE] to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
[RUN_FASTLANE] Before submitting an issue on GitHub, please follow the guide above and make
[RUN_FASTLANE] sure your project is set up correctly.
[RUN_FASTLANE] fastlane uses `xcodebuild` commands to generate your binary, you can see the
[RUN_FASTLANE] the full commands printed out in yellow in the above log.
[RUN_FASTLANE] Make sure to inspect the output above, as usually you'll find more error information there
[RUN_FASTLANE] 
[RUN_FASTLANE] [!] Error building the application - see the log above
[RUN_FASTLANE] Error: The "Run fastlane" step failed with an unknown error. Refer to "Xcode Logs" below for additional, more detailed logs.
[CLEAN_UP_CREDENTIALS] Destroying keychain - /var/folders/5p/v8tzckhs3958w1zth99dtlbc0000gp/T/eas-build-7c870c7a-0144-43d6-9f6f-781cb14c77b6.keychain
[CLEAN_UP_CREDENTIALS] Removing provisioning profile

Build failed
The "Run fastlane" step failed with an unknown error. Refer to "Xcode Logs" below for additional, more detailed logs.
npx exited with non-zero code: 1
    Error: build command failed.
error Command failed with exit code 1.
```

**Comment by tuanngocptn** [comment]
I'm stuck with this issue about 2 days. And just remove all sentry in project.pbxproj. And it work again. I know this is not the good way to fix. But I need to build and deployment in the near future. Thanks

**Comment by Taishi-Y** [comment]
run `yarn add @sentry/cli -D`

and add `@sentry/cli` to `trustedDependencies` in your package.json

```
{
  "name": "my-app",
  "version": "1.0.0",
  ...
  "trustedDependencies": ["@sentry/cli"],
  ...
}
```

It worked for me! 👍
I run this to build iOS prod build on local `eas build --profile production --platform ios --local --clear-cache --non-interactive`

**Comment by amrdraz** [comment]
@arapocket what did you do?
I'm setting up sentry for expo as well

```
"@sentry/react-native": "5.20.0",
"expo": "^50.0.17",
```


development build succeeds but production build doesn't. I get this API request failed
similar to @alterx 's log

I was thinking maybe

**Comment by horaczech** [comment]
@arapocket may I ask what alternative have you chosen? I'm also not so happy with Sentry lately

**Comment by krystofwoldrich** [comment]
Hi @alterx,
are you still experiencing the mentioned issue, could you `export SENTRY_LOG_LEVEL=debug` and share your logs in a new issue?

**Comment by krystofwoldrich** [comment]
@amrdraz Thank you for the message and the steps how you resolved it.

**Comment by krystofwoldrich** [comment]
@horaczech Could you share more about what issues are you experiencing or what parts of Sentry are you not happy with? Is there anything we can help you out with?

**Comment by lepskiy** [comment]
Hi, I have similar issue. After upgrading to Expo 50 and migrating from sentry-expo to @sentry/react-native the build (using remote EAS Build) started failing with the next error:

```
› Executing MobileAppX » Bundle React Native code and images
⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » MobileAppX/MobileAppX » Build Phases » 'Upload Debug Symbols to Sentry'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » MobileAppX/MobileAppX » Build Phases » 'Upload dSYMs to Datadog'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » MobileAppX/MobileAppX » Build Phases » '[CP-User] [RNFB] Core Configuration'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » MobileAppX/MobileAppX » Build Phases » '[CP-User] [RNFB] Crashlytics Configuration'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
▸ ** BUILD FAILED **
▸ The following build commands failed:
▸ 	PhaseScriptExecution Bundle\ React\ Native\ code\ and\ images /Users/expo/workingdir/build/ios/build/Build/Intermediates.noindex/MobileAppX.build/Debug-iphonesimulator/MobileAppX.build/Script-00DD1BFF1BD5951E006B06BC.sh (in target 'MobileAppX' from project 'MobileAppX')
▸ (1 failure)
** BUILD FAILED **
The following build commands failed:
	PhaseScriptExecution Bundle\ React\ Native\ code\ and\ images /Users/expo/workingdir/build/ios/build/Build/Intermediates.noindex/MobileAppX.build/Debug-iphonesimulator/MobileAppX.build/Script-00DD1BFF1BD5951E006B06BC.sh (in target 'MobileAppX' from project 'MobileAppX')
(1 failure)

Exit status: 65
```

These are the XCode logs:
```
    /bin/sh -c /Users/expo/workingdir/build/ios/build/Build/Intermediates.noindex/MobileAppX.build/Debug-iphonesimulator/MobileAppX.build/Script-00DD1BFF1BD5951E006B06BC.sh
+ LOCAL_NODE_BINARY=/Users/expo/.nvm/versions/node/v20.13.1/bin/node
+ '[' -z '' ']'
+ export SENTRY_PROPERTIES=sentry.properties
+ SENTRY_PROPERTIES=sentry.properties
+ '[' -z '' ']'
+ export SOURCEMAP_FILE=/Users/expo/workingdir/build/ios/build/Build/Intermediates.noindex/MobileAppX.build/Debug-iphonesimulator/MobileAppX.build/DerivedSources/main.jsbundle.map
+ SOURCEMAP_FILE=/Users/expo/workingdir/build/ios/build/Build/Intermediates.noindex/MobileAppX.build/Debug-iphonesimulator/MobileAppX.build/DerivedSources/main.jsbundle.map
+ '[' -z '' ']'
++ /Users/expo/.nvm/versions/node/v20.13.1/bin/node --print 'require('\''path'\'').dirname(require.resolve('\''@sentry/cli/package.json'\''))'
+ SENTRY_CLI_PACKAGE_PATH=/Users/expo/workingdir/build/node_modules/@sentry/cli
+ '[' -z '' ']'
+ SENTRY_CLI_EXECUTABLE=/Users/expo/workingdir/build/node_modules/@sentry/cli/bin/sentry-cli
+ REACT_NATIVE_XCODE=/Users/expo/workingdir/build/node_modules/react-native/scripts/react-native-xcode.sh
+ [[ '' == false ]]
+ ARGS=' --force-foreground --log-level=debug '
+ REACT_NATIVE_XCODE_WITH_SENTRY='"/Users/expo/workingdir/build/node_modules/@sentry/cli/bin/sentry-cli" react-native xcode  --force-foreground --log-level=debug  "/Users/expo/workingdir/build/node_modules/react-native/scripts/react-native-xcode.sh"'
+ '[' '' '!=' true ']'
+ /bin/sh -c '"/Users/expo/.nvm/versions/node/v20.13.1/bin/node" "/Users/expo/workingdir/build/node_modules/@sentry/cli/bin/sentry-cli" react-native xcode  --force-foreground --log-level=debug  "/Users/expo/workingdir/build/node_modules/react-native/scripts/react-native-xcode.sh"'
  INFO    2024-07-11 05:55:36.678262 -07:00 Loaded file referenced by SENTRY_PROPERTIES (sentry.properties)
  DEBUG   2024-07-11 05:55:36.701518 -07:00 sentry-cli version: 2.30.0, platform: "darwin", architecture: "arm64"
  INFO    2024-07-11 05:55:36.701557 -07:00 sentry-cli was invoked with the following command line: "/Users/expo/workingdir/build/node_modules/@sentry/cli-darwin/bin/sentry-cli" "react-native" "xcode" "--force-foreground" "--log-level=debug" "/Users/expo/workingdir/build/node_modules/react-native/scripts/react-native-xcode.sh"
  INFO    2024-07-11 05:55:36.701651 -07:00 Issuing a command for Organization: mpileappx Project: mpileappx
  INFO    2024-07-11 05:55:36.701658 -07:00 Using react-native build script at /Users/expo/workingdir/build/ios
  INFO    2024-07-11 05:55:36.701661 -07:00 Running in debug mode, skipping script wrapping.
+ DEST=/Users/expo/workingdir/build/ios/build/Build/Products/Debug-iphonesimulator/MobileAppX.app
+ [[ ! -n '' ]]
+ [[ Debug = *Debug* ]]
+ [[ ! iphonesimulator == *simulator ]]
+ [[ -n 1 ]]
+ echo 'SKIP_BUNDLING enabled; skipping.'
SKIP_BUNDLING enabled; skipping.
+ exit 0
+ '[' -z '' ']'
++ /Users/expo/.nvm/versions/node/v20.13.1/bin/node --print 'require('\''path'\'').dirname(require.resolve('\''@sentry/react-native/package.json'\''))'
+ SENTRY_RN_PACKAGE_PATH=/Users/expo/workingdir/build/node_modules/@sentry/react-native
+ '[' -z '' ']'
+ SENTRY_COLLECT_MODULES=/Users/expo/workingdir/build/node_modules/@sentry/react-native/scripts/collect-modules.sh
+ '[' -f /Users/expo/workingdir/build/node_modules/@sentry/react-native/scripts/collect-modules.sh ']'
+ /bin/sh /Users/expo/workingdir/build/node_modules/@sentry/react-native/scripts/collect-modules.sh
+ [[ Debug = *Debug* ]]
+ echo 'Debug build. Modules are not collected.'
Debug build. Modules are not collected.
+ exit 0
No sourcemap output has been specified.
Check that you either set a SOURCEMAP_FILE or an EXTRA_PACKAGER_ARGS environment variable in your "Bundle React Native code and images" Build Phase in XCode.
If you are not running this script from XCode, set a SOURCEMAP_FILE environment variable before running the script.
Command PhaseScriptExecution failed with a nonzero exit code
```

**Comment by arjunbanker** [comment]
Also seeing the exact same issue as @alterx, breaking on eas staging and production builds, can this issue be reopened until there is a resolution?

**Comment by charlestbell** [comment]
Happening again today.

**Comment by PhamQuyet1606** [comment]
still happens on the last version!

**Comment by felipejh** [comment]
Same here.

**Comment by bruno-garcia** [comment]
Is anyone experiencing intermittent errors with HTTP status 403?
If so, please comment/engage with:
* https://github.com/getsentry/sentry-react-native/issues/4077

We would like to solve any problem folks are having. To do that, please help us by providing some further context like: Your org slug. Times when it happened. How are you passing the org slug, auth token etc?

If your issue is not related to 403's, please raise a specific issue describing what the error is. Including debug logs and the other info I mentioned is hugely helpful for us to track down what's going on.

**Comment by arjunbanker** [comment]
yes

**Comment by krystofwoldrich** [comment]
I you are receiving 403 during the Debug Files Upload it's possible that the IP from where you are uploading the files is in range blocked by Google Cloud and so the request doesn't reach our servers.

- Details in https://github.com/getsentry/sentry-javascript-bundler-plugins/issues/605

To disable source maps auto upload, set SENTRY_DISABLE_AUTO_UPLOAD=true in your environment variables. Or to allow failing upload, set SENTRY_ALLOW_FAILURE=true. If the upload fails you will have to upload the files manually following our guides -> https://docs.sentry.io/platforms/react-native/sourcemaps/

**Comment by sergioisidoro** [comment]
I think it is quite unacceptable that a service makes uploading the source code and symbols (a non critical task) fail the entire build if the step fails. The same happens if some error like Project not found. Please check that you entered the project and organization slugs correctly.

It's effectively making thousands of build processes dependent on Sentry's service availability.

This should be a warning, not a failure, and the default should be the other way arround (eg `SENTRY_ABORT_BUILD_IF_FAIL` or something similar)

Furthermore, I am working with multiple organizations, which have different access tokens, making the secret management quite challenging.

**Comment by arjunbanker** [comment]
100%

**Comment by krystofwoldrich** [comment]
Returned to needs discussion to look at https://github.com/getsentry/sentry-react-native/issues/3551#issuecomment-2359588743

**Comment by krystofwoldrich** [comment]
Thank you @sergioisidoro,
we appreciate the feedback.

We understand that by failing the build process we are creating a dependency on Sentry. The reason is delivering the best error experience we can. That means ensuring source maps are uploaded and thus errors are symbolicated and correctly grouped into issues.

If source maps upload would be only a warning, this could cause situations where if upload would fail and the app is unknowingly release without the source map uploaded new error would be created which would not be symbolicated, this could have larger consequences like triggering alerts and more.

We always include the message `To disable source maps auto upload, set SENTRY_DISABLE_AUTO_UPLOAD=true in your environment variables. Or to allow failing upload, set SENTRY_ALLOW_FAILURE=true` to make a failed upload as smooth experience as possible.

We close this issue as the original source of the 4xx error was resolved.
